File: javascript/3740.Minimum-Distance-Between-Three-Equal-Elements-I.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minimumDistance = function (nums) {
  const map = new Map();
  let min_distance = Infinity;

  for (let i = 0; i < nums.length; i++) {
    map.set(nums[i], [...(map.get(nums[i]) || []), i]);
  }

  for (const indices of map.values()) {
    if (indices.length >= 3) {
      for (let t = 0; t <= indices.length - 3; t++) {
        const i = indices[t];
        const j = indices[t + 1];
        const k = indices[t + 2];

        const cur_distance =
          Math.abs(i - j) + Math.abs(j - k) + Math.abs(k - i);
        min_distance = Math.min(min_distance, cur_distance);
      }
    }
  }

  return min_distance === Infinity ? -1 : min_distance;
};

File: python/3740.Minimum-Distance-Between-Three-Equal-Elements-I.py
class Solution:
    def minimumDistance(self, nums: List[int]) -> int:
        index_map = defaultdict(list)
        min_distance = float('inf')

        for i, num in enumerate(nums):
            index_map[num].append(i)

        for indices in index_map.values():
            if len(indices) >= 3:
                for t in range(len(indices) - 2):
                    i, j, k = indices[t], indices[t + 1], indices[t + 2]
                    cur_distance = abs(i - j) + abs(j - k) + abs(k - i)
                    min_distance = min(min_distance, cur_distance)

        return -1 if min_distance == float('inf') else min_distance